### PR TITLE
libnetwork/iptables: some cleanups and refactoring

### PR DESCRIPTION
**Pull request by thaJeztah**
### libnetwork/iptables: inline some args

Just inline the args if they're not dynamically constructed.

### libnetwork/iptables: IPTable.NewChain() minor cleanups

- validate input variables before constructing the ChainInfo
- only construct the ChainInfo if things were successful

### libnetwork/iptables: IPTable.exists(): return early on error

Also remove a redundant string cast for the Table value.

### libnetwork/iptables: ChainInfo: don't pass whole IPTable as value

It only needed the IPVersion, so let's pass that instead.

### libnetwork/iptables: TestPassthrough(): skip without firewalld

The test was not doing anything without firewalld running, but did
not skip either.

### libnetwork/iptables: TestReloaded(): minor cleanup

- remove local bridgeName variable that shadowed the const, but
  used the same value
- remove some redundant `var` declarations, and changed fixed
  values to a const

### libnetwork/iptables: remove unused Ebtables const

This const was added in 8301dcc6d702a97feeb968ee79ae381fd8a4997a, before
being moved to libnetwork, and moved back, but it was never used.

### libnetwork/iptables: checkRunning(): use early return

Remove redundant variable declarations, and use an early return instead.

### libnetwork/iptables: merge Conn.initConnection into newConnection

initConnection was effectively just part of the constructor; ot was not
used elsewhere. Merge the two functions to simplify things a bit.

### libnetwork/iptables: signalHandler(): use s switch

It felt ever-so-slightly more readable than if/else if/(else if...)

### libnetwork/drivers/bridge: minor cleanups for iptables funcs

setupBridgeNetFiltering:
- Indicate that the bridgeInterface argument is unused (but it's needed
  to satisfy the signature).
- Return instead of nullifying the err. Still not great, but I thought it
  was very slightly more logical thing to do.

checkBridgeNetFiltering:
- Remove unused argument, and scope ipVerName to the branch where it's
  used.

### libnetwork/iptables: remove ErrIptablesNotFound

looks like this error was added in 1cbdaebaa1c2326e57945333420d25d6f77011d5,
and later moved to libnetwork in https://github.com/moby/libnetwork/commit/44c96449c2aafbb22487086ad94ff865f9ee855a
which also updated the description to something that doesn't match what
it means.

In either case, this error was never used as a special / sentinel error,
so we can just use a regular error return.

While at it, I also lower-cased the error-message; it's not string-matched
anywhere, so we can update it to make linters more happy.

### libnetwork/iptables: errors should not be capitalized

None of these errors were string-matched anywhere, so let's change them
to be non-capitalized, as they should.

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by thaJeztah** [comment]
I still have some other bits stashed locally, but I'll keep those for a follow-up.

File: libnetwork/drivers/bridge/setup_bridgenetfiltering.go [hunk 1]
```diff
@@ -31,17 +31,15 @@ func getIPVersion(config *networkConfiguration) ipVersion {
 	return ipVer
 }
 
-func setupBridgeNetFiltering(config *networkConfiguration, i *bridgeInterface) error {
-	err := checkBridgeNetFiltering(config, i)
-	if err != nil {
+func setupBridgeNetFiltering(config *networkConfiguration, _ *bridgeInterface) error {
+	if err := checkBridgeNetFiltering(config); err != nil {
 		if ptherr, ok := err.(*os.PathError); ok {
 			if errno, ok := ptherr.Err.(syscall.Errno); ok && errno == syscall.ENOENT {
```
Comment (corhere) [inline libnetwork/drivers/bridge/setup_bridgenetfiltering.go:37]: That's a very old-fashioned way of writing
```go
if errors.Is(err, syscall.ENOENT) {
```
or, if you want to be really pedantic about only matching an errno when it is wrapped in a PathError:
```go
var ptherr *os.PathError
if errors.As(err, &ptherr) && errors.Is(ptherr, syscall.ENOENT) {
```

File: libnetwork/drivers/bridge/setup_bridgenetfiltering.go [hunk 2]
```diff
@@ -31,17 +31,15 @@ func getIPVersion(config *networkConfiguration) ipVersion {
...
 					log.G(context.TODO()).Warnf("running inside docker container, ignoring missing kernel params: %v", err)
-					err = nil
-				} else {
-					err = errors.New("please ensure that br_netfilter kernel module is loaded")
+					return nil
 				}
+				err = errors.New("please ensure that br_netfilter kernel module is loaded")
 			}
 		}
 		if err != nil {
```
Comment (corhere) [inline libnetwork/drivers/bridge/setup_bridgenetfiltering.go:45]: Now that the preceding code no longer nils out `err`, this condition will always be true.

**Review by corhere — APPROVED** [review]

File: libnetwork/drivers/bridge/setup_bridgenetfiltering.go (hunk 1 above)
Comment (thaJeztah) [reply libnetwork/drivers/bridge/setup_bridgenetfiltering.go:37]: OHMAN, yes. I may include that one in my follow-up 😅

File: libnetwork/drivers/bridge/setup_bridgenetfiltering.go (hunk 2 above)
Comment (thaJeztah) [reply libnetwork/drivers/bridge/setup_bridgenetfiltering.go:45]: Will probably pick that one up as well in "the sequel" 👍

**Review by akerouanton — APPROVED** [review]
LGTM

**Pull request merged**